docs/When_Do_You_Need_A_Custom_Device.md

```diff
@@ -0,0 +1,77 @@
+It should be mentioned that VeriStand custom devices are typically not large LabVIEW applications. Custom devices are designed to be modular, self-contained plug-ins that add a specific functionality to VeriStand. While custom devices are typically developed by a single programmer, large application development best-practices may still apply. See **[LabVIEW 2018 Help: Best Practices for Large Application Development](https://zone.ni.com/reference/en-XX/help/371361R-01/lvdevconcepts/best_practices_large_apps/)** for more information.
+
+#### LabVIEW Real-Time Application Development
+
+Custom devices are typically designed to execute on real-time systems. This allows the operator to perform deterministic HIL and RT test procedures. Programming for a real-time system requires knowledge of real-time operating systems (RTOS) and specialized LabVIEW development techniques. This knowledge is typically obtained through **[NI's Training and Certification](https://www.ni.com/en-us/shop/services/education-services.html)** program by completing the **[Real-Time Application Development](https://www.ni.com/en-us/shop/services/products/labview-real-time-1-course.html)** course, and it is refined by working on several LabVIEW Real-Time applications.
+
+#### VeriStand Background
+
+Familiarity with the VeriStand Engine is crucial to successful custom device development. The correct type of custom device cannot be selected in the Custom Device Template Tool without understanding the implications of each. This knowledge is typically obtained by reading the **[VeriStand Help](https://www.ni.com/documentation/en/veristand/latest/manual/manual-overview/)**, with an emphasis on **[Understanding the VeriStand Engine](https://www.ni.com/documentation/en/veristand/latest/manual/vs-engine/)**.
+Experience with VeriStand from an operator's perspective is highly desired. This experience enables you to build operator-friendly interfaces that conform to the standard look and feel of other VeriStand components. Familiarity with VeriStand allows the developer to build-up a complex system definition, which allows thorough and realistic testing and benchmarking.
+
+#### Hardware Driver Development
+
+Custom device must call a hardware or instrument driver to support 3rd-party hardware. All National Instruments hardware comes with a LabVIEW Application Program Interface (API) that can be used in the custom device. However, just because a LabVIEW API exists does not guarantee the custom device can be easily implemented. Consider the following points when evaluating the feasibility of a custom device for 3rd-party hardware.
```

Comment:
```suggestion
The custom device must call a hardware or instrument driver to support 3rd-party hardware. All National Instruments hardware comes with a LabVIEW Application Program Interface (API) that can be used in the custom device. However, just because a LabVIEW API exists does not guarantee the custom device can be easily implemented. Consider the following points when evaluating the feasibility of a custom device for 3rd-party hardware.
```